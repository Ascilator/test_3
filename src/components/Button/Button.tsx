type ButtonProps = {
  text: string;
  onClick: () => void;
  className?: string;
};

export const Button: React.FC<ButtonProps> = ({ text, className, onClick }) => {
  return (
    <button
      className={`px-5 py-3 border border-cyan-500 bg-cyan-500 rounded-xl text-white ${className}`}
      onClick={onClick}
    >
      {text}
    </button>
  );
};
