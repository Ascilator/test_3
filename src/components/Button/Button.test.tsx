import React from "react";
import { Button } from "./Button";
import { fireEvent, render, screen } from "@testing-library/react";

describe("Button test", () => {
  test("Button is rendering", () => {
    const { container } = render(
      <Button text="text" onClick={() => {}}></Button>
    );
    expect(container).toMatchSnapshot();
  });

  test("Button is clicking and displaing in the correct way", () => {
    const handleClick = jest.fn();

    const { getByText } = render(
      <Button text="text" onClick={handleClick}></Button>
    );

    const buttonElement = getByText("text");
    fireEvent.click(buttonElement);

    expect(getByText("text")).toBeTruthy();
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
