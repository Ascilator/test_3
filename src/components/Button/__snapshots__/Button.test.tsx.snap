// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test Button is rendering 1`] = `
<div>
  <button
    class="px-5 py-3 border border-cyan-500 bg-cyan-500 rounded-xl text-white undefined"
  >
    text
  </button>
</div>
`;
